homeassistant/components/modbus/fan.py

```diff
@@ -0,0 +1,71 @@
+"""Support for Modbus fans."""
+import logging
+from typing import Any, Dict, Optional
+
+from homeassistant.components.fan import FanEntity
+from homeassistant.const import CONF_NAME
+from homeassistant.helpers.typing import (
+    ConfigType,
+    DiscoveryInfoType,
+    HomeAssistantType,
+)
+
+from .const import CALL_TYPE_COIL, CONF_FANS, CONF_INPUT_TYPE, MODBUS_DOMAIN
+from .modbus import ModbusHub
+from .switch import ModbusCoilSwitch, ModbusRegisterSwitch
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(
+    hass: HomeAssistantType,
+    config: ConfigType,
+    async_add_entities,
+    discovery_info: Optional[DiscoveryInfoType] = None,
+):
+    """Read configuration and create Modbus fan."""
+    if discovery_info is None:
+        return
+
+    fans = []
+    for entry in discovery_info[CONF_FANS]:
+        hub: ModbusHub = hass.data[MODBUS_DOMAIN][discovery_info[CONF_NAME]]
+        if entry[CONF_INPUT_TYPE] == CALL_TYPE_COIL:
+            fans.append(ModbusCoilFan(hub, entry))
+        else:
+            fans.append(ModbusRegisterFan(hub, entry))
+    async_add_entities(fans)
+
+
+class ModbusCoilFan(ModbusCoilSwitch, FanEntity):
+    """Representation of a Modbus coil fan."""
+
+    def __init__(self, hub: ModbusHub, config: Dict[str, Any]):
```

Comment:
Looks like this __init__ can be safely removed

Reply:
I'm not a Python guru, but looks like this would be enough:
```
class ModbusCoilFan(ModbusCoilSwitch, FanEntity):
    """Representation of a Modbus coil fan."""
    pass 
```
And codecov would be pretty happy about it.

Reply:
You're right. I was thinking to make the code more explicit and to increase readibility. But when I thought about this, unit tests should guarantee my code works as expected. I will simplify as you suggest.